ecl/hthor/hthor.cpp

```diff
@@ -1601,7 +1610,16 @@ class CHThorPipeThroughActivity : public CHThorSimpleActivityBase, implements IP
         // called from puller thread
         if(recreate)
             openPipe(helper.getNameFromRow(row));
-        writeTransformer->writeTranslatedText(row, pipe);
+        try
+        {
+            writeTransformer->writeTranslatedText(row, pipe);
+        }
+        catch (IException *e)
+        {
+            pipeException.set(e);
+            ReleaseRoxieRow(row);
+            throw e;
```

Comment:
If we are throwing, why set pipeException ?

Reply:
I'm looking at what changes would be involved (and how it would simply handling in this code) if a IPipeProcessException type was introduced.